tests/visualization_tests/matplotlib_tests/test_optimization_history.py

```diff
@@ -1,236 +1,26 @@
-    assert len(figure.get_lines()) == 3
-    xydata = figure.collections[1].get_offsets()
-    assert np.array_equal(xydata[:, 0], [0.0, 1.0, 2.0])
-    assert np.array_equal(xydata[:, 1], [0.0, 1.0, 2.0])
-    plt.savefig(BytesIO())
-
-    # Test customized target name.
-    custom_target_name = "Target Name"
-    figure = plot_optimization_history(studies, target_name=custom_target_name, error_bar=True)
-    legend_texts = [legend.get_text() for legend in figure.legend().get_texts()]
-    assert sorted(legend_texts) == ["Best Value", custom_target_name]
-    assert figure.get_ylabel() == custom_target_name
-    plt.savefig(BytesIO())
-
-    # Ignore failed trials.
-    studies = [create_study(direction=direction) for _ in range(n_studies)]
-    for study in studies:
-        study.optimize(fail_objective, n_trials=1, catch=(ValueError,))
-
-    figure = plot_optimization_history(studies, error_bar=True)
-    assert len(figure.get_lines()) == 0
-    plt.savefig(BytesIO())
-
-
-@pytest.mark.parametrize("direction", ["minimize", "maximize"])
-def test_error_bar_in_optimization_history(direction: str) -> None:
-    def objective(trial: Trial) -> float:
-        return trial.suggest_float("x", 0, 1)
-
-    studies = [create_study(direction=direction) for _ in range(3)]
-    suggested_params = [0.1, 0.3, 0.2]
-    for x, study in zip(suggested_params, studies):
-        study.enqueue_trial({"x": x})
-        study.optimize(objective, n_trials=1)
-    figure = plot_optimization_history(studies, error_bar=True)
-    mean = np.mean(suggested_params)
-    std = np.std(suggested_params)
-
-    np.testing.assert_almost_equal(figure.get_lines()[0].get_ydata(), mean)
-    np.testing.assert_almost_equal(figure.get_lines()[1].get_ydata(), mean - std)
-    np.testing.assert_almost_equal(figure.get_lines()[2].get_ydata(), mean + std)
+from optuna.visualization._optimization_history import _OptimizationHistoryInfo
+from optuna.visualization.matplotlib._matplotlib_imports import plt
+from optuna.visualization.matplotlib._optimization_history import _get_optimization_history_plot
+from tests.visualization_tests.test_optimization_history import optimization_history_info_lists
+
+
+@pytest.mark.parametrize("target_name", ["Objective Value", "Target Name"])
+@pytest.mark.parametrize("info_list", optimization_history_info_lists)
+def test_get_optimization_history_plot(
+    target_name: str, info_list: List[_OptimizationHistoryInfo]
+) -> None:
+    figure = _get_optimization_history_plot(info_list, target_name=target_name)
+    assert figure.get_ylabel() == target_name
+    expected_legends = []
+    for info in info_list:
+        expected_legends.append(info.values_info.label_name)
+        if info.best_values_info is not None:
+            expected_legends.append(info.best_values_info.label_name)
+    legends = [legend.get_text() for legend in figure.legend().get_texts()]
+    assert sorted(legends) == sorted(expected_legends)
     plt.savefig(BytesIO())
```

Comment:
If we implement the test for saving a figure in `visualization_tests/test_optimization_history.py`, we don't need this line anymore.

```suggestion
```

Reply:
Thanks for the comment! Is there any reason for moving the test of matplotlib to `visualization_tests/test_optimization_history.py` while there exists `matplotlib_tests/` directory? How about discussing the consistency of test file style in another issue?
https://github.com/optuna/optuna/pull/3806#discussion_r931725918

Reply:
Ah, I got the point now, thank you for your explanation. The current form makes sense to avoid introducing backend-based conditions, which are not readable. Please ignore my suggestions 🙇🏻

Reply:
Yeah, I feel the conditional branch by the returned type makes the test complex and might miss some bugs such as matplotlib's plot function returning `go.Figure` object (this example might not be realistic though). Anyway, let's discuss this probably after all the PRs on test refactoring are merged 👍